### PR TITLE
Update the max TP threads count to sys.maxsize

### LINKED ISSUE #873: Allow specifying PYTHON_THREADPOOL_THREAD_COUNT larger than 32
Currently, the worker limits the maximum value of `PYTHON_THREADPOOL_THREAD_COUNT` to 32. Any value larger than that is ignored. There is no good reason for this in general case: if someone/something decided a larger number works better for a specific use case, this should be respected.

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
With new work being done on the Host side control concurrency, the 32 thread number isn't required. Setting the max value to 1024.

Fixes #873

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure_functions_worker/dispatcher.py
Comment (AnatoliB) [inline azure_functions_worker/dispatcher.py:571]: I still don't understand why we even have this check at all. Regardless of the MAX value, what is the reason?

As a user, I expect the value I specify to be either accepted (preferred) or at least explicitly rejected (if there is an actual reason). Right now, if I specify 1025, this value will not be accepted and, even worse, the worker will fall back to the very low default, and I may not notice the warning. And, _in order to achieve these troubles_, we have quite a lot of code!

If we didn't have this check at all, what would be the motivation for us to add it? Are we assuming we know more than the user who specified a certain value? How can we know that?

**Review by Hazhzeng — APPROVED** [review]

File: azure_functions_worker/dispatcher.py
Comment (vrdmr) [reply azure_functions_worker/dispatcher.py:571]: After offline conversation - we have decided to remove any artificial limits on the maximum number of threads.

**Review by AnatoliB — APPROVED** [review]
I would even recommend _removing_ the code performing this check, including the unit tests and comments, as this is effectively dead code now. But I'll leave this up to you. Other than that, LGTM and thank you!

**Comment by vrdmr** [comment]
@AnatoliB - Yes, that's the plan - to remove the code and the unit tests. I wanted to get this change in sooner to unblock the perf testing work and get this part of the next week's release.

**Pull request merged**
